### DISCUSSION TITLE
PCB design experience?

### DISCUSSION THREAD

**Discussion by netmindz** [netmindz/balboa_GL_ML_spa_control#29] [General]
Now the requirements are better known, it would be great if someone with the right skills were able to design a single PCB that has the 9v to 5v PSU, RS485 adapter etc all on one PCB, so that all the user needs to do is attach the mini molex connector and they are good to go

**Comment by jjacobsen-snake** [comment]
(Edit! New version)
I am currently trying out a custom PCB. Although I don't need the HA/MQTT from your code. I was simply thinking of using Arduino IOT Cloud. But you could certainly use this.
v1.00...
![V1.00](https://github.com/netmindz/balboa_GL_ML_spa_control/assets/66822174/e551de6d-53a5-4887-a28d-4895054d4115)
v1.01...
![V1.01](https://raw.githubusercontent.com/jjacobsen-snake/GL2000-ESP32/main/3D-Renderings/v1.01/GL2000-ESP32-3.jpg)

**Reply by netmindz** [reply]
Not sure I have space for PCB to plug into main board like that, but could always use extension cable.
Looks nice compact design

**Reply by netmindz** [reply]
Link to the details for it?

**Reply by netmindz** [reply]
@jjacobsen-snake ?

**Reply by jjacobsen-snake** [reply]
Was on vacation and tried it, unfortunately it had a shorted BSS138 mosfet feeding the 5v to the ESP32. I have a 3.5+hour drive to my spa under test, so might be some time. The card sits off-center, so it would not interfere with other connectors. I could ship you one card if you were interested? I will try to create a github upload for this card, as I would like to contribute.

**Reply by jjacobsen-snake** [reply]
Added the current design to https://github.com/jjacobsen-snake/GL2000-ESP32

**Reply by davewatson91** [reply]
wow, this looks great. Also don't have space inside my enclosure though - would need to do a molex extension and a case for it. I'm assuming the USB is for serial only, and not for power? 

Did the mosfet short from a design flaw, or something else?

**Reply by jjacobsen-snake** [reply]
Interesting, I thought there would be enough room for this anywhere. Welp, at least the Balboa GL2000 had. Is there some changes I could make to make it easier/more flexible to fit? The USB is just for convenience, it doubles as programming/UART, power and debug. So for programming you only need a USB-C and this board, not the entire spa. The board can draw power from the spa-connector and/or USB.
The BSS138 I had were used, never bothered measuring them before installation sadly. I've ordered new ones from mouser.

**Reply by davewatson91** [reply]
Love the idea of the USB-C - it's absolutely brilliant. 

My GL2000 is in a plastic enclosure under the spa, which is maybe only 80mm deep. I've attached a couple of photos to try and show where it would foul. 

![IMG_3854](https://github.com/netmindz/balboa_GL_ML_spa_control/assets/51277302/98285956-0daa-4027-9375-bbd2abecbe3b)
![IMG_3852](https://github.com/netmindz/balboa_GL_ML_spa_control/assets/51277302/b4b598fe-d77a-40a0-ac76-0d832a946441)

**Reply by tmjo** [reply]
Mine looks the same as yours! Think the safest solution is to have an external box with molex-extender to it.

**Reply by netmindz** [reply]
Yeah I have same enclosure. Can't remember who it was that posted a pic of a metal enclosure

**Reply by jjacobsen-snake** [reply]
Exactly the same as the one we have, it plugs in nicely in the top connector and tucks under the incoming cables on the right side. Check out the PCB size on this image, if you have 45mm above the top connector (think it's called J73) then you're good. :)

I will order up some parts for the voltage regulators and I will let you guys know when they arrive.
 
 
![PCB-Size](https://github.com/netmindz/balboa_GL_ML_spa_control/assets/66822174/c84dc8ff-963b-48a5-aa61-b52bf03b6d8d)

**Reply by davewatson91** [reply]
Wow, that's insanely smaller than I anticipated. I've just measured mine, and it should clear without an issue at all - I have J72 at the top, but assuming that it's either just a difference in the original PCB print or something similar. 

How much are you wanting for these? Happy to send through some money whenever you'd like - this will make an insanely neater install than I would otherwise be doing! (even if I 3D print an enclosure, I'm still running additional cables, multiple pcb's and such.

**Reply by netmindz** [reply]
Have you v1.1 boards arrived now @jjacobsen-snake ?

**Comment by netmindz** [comment]
Would be awesome to get hold of one of you have a spare.

If you ping me an email I can send my address will@netmindz.net

I have a spare male cable and female to female cable I can use to make an extension if there is any space issues in my GL2001
